core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -110,6 +204,26 @@ private[zio] final class FiberContext[E, A](platform: Platform, startEnv: AnyRef
     // Put the maximum operation count on the stack for fast access:
     val maxopcount = executor.yieldOpCount
 
+    // Put tracing configuration on the stack:
+//    val traceExec    = this.traceExec
+//    val traceStack   = this.traceStack
+//    val traceEffects = this.traceEffects
+
+    // Store the trace of the immediate future flatMap during evaluation
+    // of a 1-hop left bind, to show a stack trace closer to the point of failure
+    var fastPathFlatMapContinuationTrace: ZTraceElement = null
```

Comment:
This will be allocated on the heap in an `ObjectRef`.

Reply:
Which argues for pulling this logic out into the aforementioned utility class.

Reply:
Are you sure? I'm assigning to a val before passing. Also null can't be allocated on the heap?

Reply:
Ok, I see that

Reply:
inlining forces that to disappear, but gives, consistently, just slightly lower performance:
`@noinline` with ObjectRef:
```
[info] Result "scalaz.zio.IOLeftBindBenchmark.scalazLeftBindBenchmark":
[info]   4930.188 ±(99.9%) 77.862 ops/s [Average]
[info]   (min, avg, max) = (4817.133, 4930.188, 4987.029), stdev = 51.501
[info]   CI (99.9%): [4852.325, 5008.050] (assumes normal distribution)
```
`@inline` without ObjectRef:
```
[info] Result "scalaz.zio.IOLeftBindBenchmark.scalazLeftBindBenchmark":
[info]   4893.806 ±(99.9%) 74.707 ops/s [Average]
[info]   (min, avg, max) = (4821.052, 4893.806, 4989.764), stdev = 49.414
[info]   CI (99.9%): [4819.099, 4968.512] (assumes normal distribution)
```
'worker-wrapper' with `@noinline`  on part of fastPathTrace and inlined assignment without ObjectRef
```
[info] Result "scalaz.zio.IOLeftBindBenchmark.scalazLeftBindBenchmark":
[info]   4887.192 ±(99.9%) 105.314 ops/s [Average]
[info]   (min, avg, max) = (4777.507, 4887.192, 5003.847), stdev = 69.658
[info]   CI (99.9%): [4781.878, 4992.505] (assumes normal distribution)
```

Seems like the gain from `@noinline` hot-spot is consistently higher here than the overhead of ObjectRef